packages/patternfly-3/patternfly-react/src/components/VerticalNav/VerticalNavItemHelper.js

```diff
@@ -82,18 +82,15 @@ class BaseVerticalNavItemHelper extends React.Component {
 
   onItemClick = event => {
     const { primary, secondary, tertiary } = this.getContextNavItems();
-    const { isMobile, updateNavOnItemClick, idPath } = this.props;
-    const { href, onClick } = this.navItem();
-    event.preventDefault();
+    const { isMobile, preventHref, updateNavOnItemClick, idPath } = this.props;
+    const { onClick } = this.navItem();
+    if (preventHref && !!onClick) event.preventDefault();
     updateNavOnItemClick(primary, secondary, tertiary, this.idPath(), idPath); // Clears all mobile selections
     if (isMobile) {
       this.onMobileSelection(primary, secondary, tertiary); // Applies new mobile selection here
     }
     this.setActive();
     onClick && onClick(primary, secondary, tertiary);
-    if (href) {
-      window.location = href; // Note: This should become router-aware later on.
-    }
   };
```

Comment:
Should this remain when not preventHref?

Reply:
seems unnecessary because the `<a>` has the `href` now